extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/KafkaStreamsTopologyManager.java

```diff
@@ -1,282 +1,29 @@
 package io.quarkus.kafka.streams.runtime;
 
-import java.net.InetSocketAddress;
-import java.time.Duration;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.HashSet;
-import java.util.List;
-import java.util.Objects;
-import java.util.Optional;
 import java.util.Properties;
 import java.util.Set;
 import java.util.concurrent.ExecutionException;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
-import java.util.function.Function;
-import java.util.stream.Collectors;
 
-import javax.enterprise.context.ApplicationScoped;
-import javax.enterprise.event.Observes;
-import javax.enterprise.inject.Instance;
-import javax.enterprise.inject.Produces;
-import javax.inject.Inject;
-import javax.inject.Singleton;
-
-import org.apache.kafka.clients.CommonClientConfigs;
 import org.apache.kafka.clients.admin.AdminClient;
-import org.apache.kafka.clients.admin.AdminClientConfig;
 import org.apache.kafka.clients.admin.ListTopicsResult;
-import org.apache.kafka.common.config.SaslConfigs;
-import org.apache.kafka.common.config.SslConfigs;
-import org.apache.kafka.streams.KafkaClientSupplier;
-import org.apache.kafka.streams.KafkaStreams;
-import org.apache.kafka.streams.KafkaStreams.StateListener;
-import org.apache.kafka.streams.StreamsConfig;
-import org.apache.kafka.streams.Topology;
-import org.apache.kafka.streams.processor.StateRestoreListener;
 import org.jboss.logging.Logger;
 
-import io.quarkus.runtime.ShutdownEvent;
-import io.quarkus.runtime.StartupEvent;
-
-/**
- * Manages the lifecycle of a Kafka Streams pipeline. If there's a producer
- * method returning a KS {@link Topology}, then this topology will be configured
- * and started. Optionally, before starting the pipeline, this manager will wait
- * for a given set of topics to be created, as KS itself will fail without all
- * input topics being created upfront.
- */
-@ApplicationScoped
 public class KafkaStreamsTopologyManager {
 
     private static final Logger LOGGER = Logger.getLogger(KafkaStreamsTopologyManager.class.getName());
 
-    private final ExecutorService executor;
-    private volatile KafkaStreams streams;
-    private volatile KafkaStreamsRuntimeConfig runtimeConfig;
-    private volatile Instance<Topology> topology;
-    private volatile Properties properties;
     private volatile Properties adminClientConfig;
```

Comment:
This looks like it can be `final` now, correct?

Reply:
Ah yeah, forgot to remove it.

Reply:
Fixed. Removed `volatile` and added `final`.